pkg/handler/scale_handler.go

```diff
@@ -337,21 +337,31 @@ func (h *ScaleHandler) updateDeployment(deployment *apps_v1.Deployment) error {
 	return err
 }
 
-func (h *ScaleHandler) resolveEnv(deployment *apps_v1.Deployment) (map[string]string, error) {
+func (h *ScaleHandler) resolveEnv(deployment *apps_v1.Deployment, containerName string) (map[string]string, error) {
 	deploymentKey, err := cache.MetaNamespaceKeyFunc(deployment)
 	if err != nil {
 		return nil, err
 	}
 
 	if len(deployment.Spec.Template.Spec.Containers) < 1 {
 		return nil, fmt.Errorf("Deployment (%s) doesn't have containers", deploymentKey)
-	} else if len(deployment.Spec.Template.Spec.Containers) > 1 {
-		return nil, fmt.Errorf("Deployment (%s) has more than one container", deploymentKey)
 	}
 
-	container := deployment.Spec.Template.Spec.Containers[0]
 	resolved := make(map[string]string)
 
+	var container core_v1.Container
+
+	if containerName != "" {
+		for _, c := range deployment.Spec.Template.Spec.Containers {
+			if c.Name == containerName {
+				container = c
+				break
+			}
+		}
+	} else {
+		container = deployment.Spec.Template.Spec.Containers[0]
+	}
```

Comment:
Handle a nil container of the specified containerName doesn’t exist.

Reply:
updated